libdevcore/JsonUtils.h

```diff
@@ -24,4 +24,25 @@ namespace dev
 {
 // Throws UnknownField() if _obj contains field names not listed in _allowedFields.
 void validateFieldNames(json_spirit::mObject const& _obj, std::set<std::string> const& _allowedFields);
+
+// Converts json value type to string
+std::string jsonTypeAsString(json_spirit::Value_type _type);
+
+enum jsonField
+{
+    Required,
+    Optional
+};
+using jsonTypeSet = std::set<json_spirit::Value_type>;
+using jsonType = std::pair<jsonTypeSet, jsonField>;
+//! Check the json object with validation map that reuires certain field of certain type to be
```

Comment:
Our coding style prescribes @notation rather than \notation in doxygen comments (that is, change the following lines to use `@param`)
Also the rest of the code tends to mark doxygen comment with triple slash `///` instead of `//!`

Though to be honest I don't see much point in using doxygen comments at all for us.

Reply:
Actually sometimes there's `/**` too, but I don't remember many `//!` in our code